### PR TITLE
Cds 45148

### PR DESCRIPTION
**Pull request by michaelcretzman**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ x] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* https://harness.atlassian.net/browse/CDS-45148

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot. 

[CDS-45148]: https://harness.atlassian.net/browse/CDS-45148?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by douglas-j-bothwell — APPROVED** [review]
A few sugs but otherwise looks good!

Exceptions save you the time of having to select multiple subordinate entities individually. 
^ Suggest this rewording:  Exceptions save you time, since you don't need to select multiple subordinate entries individually. 

Deployment Freeze is a CD feature only. It does not apply to other module stages like CI and Feature Flags.
^ Perhaps make this a note?

File: docs/continuous-delivery/cd-deployments-category/deployment-freeze.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+      enabled: true
+```
+```mdx-code-block
+  </TabItem>
+</Tabs>
+```
+
+## Notify users of freeze window events
+
+You can notify Harness users and people outside of your Harness account using freeze window notifications.
```
Comment (ishantmehtaharness) [inline docs/continuous-delivery/cd-deployments-category/deployment-freeze.md:217]: Not sure, if this is possible today
Posted on slack: https://harness.slack.com/archives/C03M81KT7K4/p1672906818578569

**Review by ishantmehtaharness — DISMISSED** [review]

File: docs/continuous-delivery/cd-deployments-category/deployment-freeze.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+## Deployment freeze summary
+
+If you are new to deployment freezes, review the following summary.
+
+<details>
+<summary>Deployment freeze summary</summary>
+
+A deployment freeze is a period of time during in which no new changes are made to a system or application. This ensures that a system or application remains stable and free of errors, particularly in the lead-up to a major event or release.
+
+During a deployment freeze, only critical bug fixes and security patches might be deployed, and all other changes are put on hold until the freeze is lifted.
```
Comment (udayvunnam) [inline docs/continuous-delivery/cd-deployments-category/deployment-freeze.md:22]: `During a deployment freeze, only critical bug fixes and security patches might be deployed`. This is not correct at the moment. This doesn't indicate how critical bug fixes and security patches might be deployed during freeze

File: docs/continuous-delivery/cd-deployments-category/deployment-freeze.md [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+
+Deployment freezes are commonly used in software development to ensure that a system is not destabilized by the introduction of new code in new application versions.
```
Comment (udayvunnam) [inline docs/continuous-delivery/cd-deployments-category/deployment-freeze.md:24]: Can we make `Deployment freezes` -> `Deployment freeze`? I see mixed usage currently

File: docs/continuous-delivery/cd-deployments-category/deployment-freeze.md [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+
+For each scope level, you can select all subordinate entities and then add exceptions. For example, you can select **All Projects** at the org level, but then select one or more projects as exceptions.
+
+Exceptions save you the time of having to select multiple subordinate entities individually.
+
+### What about pipelines already running?
+
+If a pipeline is running and a freeze happens, the pipeline will contine to run until the current stage of the pipeline has executed. Once that stage executes, the freeze is implemented and no further stages will execute.
+
+If the freeze happens to a running pipeline and it is unable to complete all stages, the status of the pipeline execution is listed as **Aborted With Freeze**.
```
Comment (udayvunnam) [inline docs/continuous-delivery/cd-deployments-category/deployment-freeze.md:58]: Aborted With Freeze -> Aborted By Freeze

File: docs/continuous-delivery/cd-deployments-category/deployment-freeze.md (hunk 1 above)
Comment (ishantmehtaharness) [reply docs/continuous-delivery/cd-deployments-category/deployment-freeze.md:217]: Checked with @shylajas , its possible

File: docs/continuous-delivery/cd-deployments-category/deployment-freeze.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+
+If you have set up a deployment freeze on an account, org, or project, you cannot initiate deployment on the frozen entities during the freeze schedule.
+
+### Access control
+
+Deployment freeze access control is configured using the **Deployment Freeze** role permissions.
+
+- **Manage**: add/edit/delete freeze at any level.
+- **Override**: When a deployment is required during a freeze duration, users with this role can still perform deployments.
+- **Global**: enable/disable freeze at account level.
```
Comment (udayvunnam) [inline docs/continuous-delivery/cd-deployments-category/deployment-freeze.md:86]: This is to manage Global Freeze at all levels which is different from account-level freeze

File: docs/continuous-delivery/cd-deployments-category/deployment-freeze.md [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+
+   The coverage will look something like this:
+
+   ![coverage](static/deployment-freeze-coverage.png)
+
+5. Click **Continue**.
+
+   In **Schedule**, you define when the freeze windows starts and stops.
+
+6. In **Timezone**, select a timezone.
```
Comment (udayvunnam) [inline docs/continuous-delivery/cd-deployments-category/deployment-freeze.md:142]: Auto selected as user timezone. Change if required

File: docs/continuous-delivery/cd-deployments-category/deployment-freeze.md [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+7. In **Start Time**, select a calendar date and time for the freeze window to start.
+8. In **End Time**, select a duration (for example `1d`) or an end date and time.
```
Comment (udayvunnam) [inline docs/continuous-delivery/cd-deployments-category/deployment-freeze.md:144]: minimum duration (start-end) is 30m we can mention it somewhere

**Review by thisrohangupta — APPROVED** [review]
If there is no API available then I approve, but we need to capture this an enhancement to the documentation.

**Pull request merged**